deepobs/tensorflow/datasets/dataset.py

```diff
@@ -44,8 +44,9 @@ def __init__(self, batch_size):
 
         # Reinitializable iterator given types and shapes of the outputs
         # (needs to be the same for train and test of course).
-        self._iterator = tf.data.Iterator.from_structure(
-            self._train_dataset.output_types, self._train_dataset.output_shapes
+        self._iterator = tf.compat.v1.data.Iterator.from_structure(
+            tf.compat.v1.data.get_output_types(self._train_dataset),
+            tf.compat.v1.data.get_output_shapes(self._train_dataset),
         )
         self.batch = self._iterator.get_next()
```

Comment:
This throws the error

```txt
 GetNext() failed because the iterator has not been initialized. Ensure that you have run the initializer operation for this iterator before getting the next element. [Op:IteratorGetNext]
 ```

 It appears that `self.dataset.batch` is accessed in `set_up` of (at least one - `cifar100_3c3d`) `TestProblem` after initialization. The question is which initialization to choose then I guess.